homeassistant/components/media_player/channels.py

```diff
@@ -0,0 +1,395 @@
+            self.channel_number = None
+            self.channel_name = None
+            self.channel_image_url = None
+
+        if np_hash:
+            self.now_playing_title = np_hash.get('title', None)
+            self.now_playing_episode_title = np_hash.get('episode_title', None)
+            self.now_playing_season_number = np_hash.get('season_number', None)
+            self.now_playing_episode_number = np_hash.get('episode_number',
+                                                          None)
+            self.now_playing_summary = np_hash.get('summary', None)
+            self.now_playing_image_url = np_hash.get('image_url', None)
+        else:
+            self.now_playing_title = None
+            self.now_playing_episode_title = None
+            self.now_playing_season_number = None
+            self.now_playing_episode_number = None
+            self.now_playing_summary = None
+            self.now_playing_image_url = None
+
+    @property
+    def name(self):
+        """Return the name of the player."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the player."""
+        if self.status == 'stopped':
+            return STATE_IDLE
+
+        if self.status == 'paused':
+            return STATE_PAUSED
+
+        if self.status == 'playing':
+            return STATE_PLAYING
+
+        return STATE_UNKNOWN
+
+    def update(self):
+        """Retrieve latest state."""
+        self.update_favorite_channels()
+        self.update_state(self.client.status())
+
+    @property
+    def source_list(self):
+        """List of favorite channels."""
+        sources = [channel['name'] for channel in self.favorite_channels]
+        return sources
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self.muted
+
+    @property
+    def media_content_id(self):
+        """Content ID of current playing channel."""
+        return self.channel_number
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_CHANNEL
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        if self.now_playing_image_url:
+            return self.now_playing_image_url
+        elif self.channel_image_url:
+            return self.channel_image_url
+        else:
+            return 'https://getchannels.com/assets/img/icon-1024.png'
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        if self.state == STATE_UNKNOWN:
+            return "Offline"
+        else:
+            return self.now_playing_title
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return FEATURE_SUPPORT
+
+    def mute_volume(self, mute):
+        """Mute (true) or unmute (false) player."""
+        if mute != self.muted:
```

Comment:
This is assuming the state is up to date which might not be the case.

Reply:
There's no discrete mute though, just a toggle :/